src/combinatorics/simple_graph/ends/defs.lean

```diff
@@ -0,0 +1,256 @@
+/-
+Copyright (c) 2022 Anand Rao, Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anand Rao, Rémi Bottinelli
+-/
+import combinatorics.simple_graph.basic
+import combinatorics.simple_graph.connectivity
+import data.set_like.basic
+import category_theory.category.basic
+import category_theory.filtered
+import topology.category.Top.limits
+
+/-!
+# Ends
+
+This file contains a definition of the ends of a simple graph, as sections of the inverse system
+assigning, to each finite set of vertices, the connected components of its complement.
+-/
+
+universes u
+variables {V : Type u} (G : simple_graph V) (K L L' M : set V)
+
+namespace simple_graph
+
+section out
+
+/-- The graph induced by removing `K` -/
+@[reducible] def out := G.induce $ Kᶜ
+
+/-- Subsetship induces an obvious map on the induced graphs. -/
+@[reducible] def out_hom {K L} (h : K ⊆ L) : G.out L →g G.out K :=
+{ to_fun := λ v, ⟨v.val, set.compl_subset_compl.mpr h v.prop⟩,
+  map_rel' := by { rintros ⟨_, _⟩ ⟨_, _⟩, exact id } }
```

Comment:
Can this be a general definition about `simple_graph.induce`? Then I think you can avoid defining `simple_graph.out` entirely.

Reply:
Thanks, that's done and better. Would you argue for getting rid of `comp_out` too, somehow? or keeping it but moving its lemmas only to more general constructions?